src/Node/Expression/InlinePrint.php

```diff
@@ -0,0 +1,35 @@
+<?php
+
+/*
+ * This file is part of Twig.
+ *
+ * (c) Fabien Potencier
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Twig\Node\Expression;
+
+use Twig\Compiler;
+use Twig\Node\Node;
+
+/**
+ * @internal
+ */
+final class InlinePrint extends AbstractExpression
+{
+    public function __construct(Node $node, $lineno)
+    {
+        parent::__construct(['node' => $node], [], $lineno);
+    }
+
+    public function compile(Compiler $compiler)
+    {
+        $compiler
+            ->raw('print (')
```

Comment:
aren't we using `echo` in compiled templates ?

Reply:
because echo does not work here :)

Reply:
hmm, indeed. echo can only start a statement (as does `throw`), while print can be used in expressions. I forgot that they are not just aliases (I haven't used print since at least 7 years :smile:).